src/case_mapping.rs

```diff
@@ -33,6 +35,15 @@ pub fn to_titlecase(chr: char) -> [u32; 3] {
     lookup(chr as u32).2
 }
 
+/// Map the supplied character to its case-folded equivalent.
+///
+/// **Note:** A result of zero indicates the codepoint maps to itself.
+pub fn case_folded(chr: char) -> u32 {
```

Comment:
I'm not sure about mapping to plain u32 instead of Option<NonZeroU32>; it's likely less nice to use but it's consistent with other methods which seemed maybe more important?